### ISSUE TITLE
List plugins by category

### ISSUE THREAD

**Issue by jennydaman** [FNNDSC/ChRIS_store_ui#137]
![screenshot](https://user-images.githubusercontent.com/20404439/114314038-1f589300-9ac7-11eb-8755-8fcc69295ea2.png)

The left sidebar showing Categories is currently filler.

The backend can be queried for plugins belonging to a certain category.

```
https://chrisstore.co/api/v1/plugins/search/?category=Segmentation
```

Considerations: can the backend be queried for a list of categories?
For the time being, it might be better to hard-code the list of categories into the front-end.

**Comment by zrthxn** [comment]
I'd like to start working on this.

**Comment by zrthxn** [comment]
I wanted to ask, since you mentioned in another issue that you were looking to move towards functional components, would it be okay if parts of the plugin list component were ported to functional components? Since filtering would be a new feature to add, maybe it'd make sense to revamp the component along the way?

**Comment by jennydaman** [comment]
That was mentioned by @PintoGideon in a different repo: https://github.com/FNNDSC/ChRIS_ui/issues/206

I have used the word "functional" in passing, meaning "working," not the JSX jargon. Though I suppose it'd be an upgrade.

Regarding revamping, the effort is happening in the issue https://github.com/FNNDSC/ChRIS_store_ui/issues/101 and being spearheaded by @DivyanshiSingh. If you are concerned about refactors, beautiful code, and doing things for the long run, consider making PRs to the `pf4` branch instead.

**Comment by zrthxn** [comment]
I've found that instead of querying the backend for plugins in a category its simpler to filter by category on the frontend since I'm still not sure if the backend can be queried for a list of categories (after reading the docs for a while).

For the list of categories, I'm accumulating the list from the plugins which are fetched when the page loads.

**Comment by jennydaman** [comment]
@zrthxn In the first comment, I gave an example for how to query the backend API for plugins belonging to a given category. The example is reproduced below:

```
https://chrisstore.co/api/v1/plugins/search/?category=Segmentation
```

Reading the docs is usually not productive because, well, there's not much. It is shameful to say this, but the best way to understand the API is to:

1. ask me
2. try things out

As I've explained in https://github.com/FNNDSC/ChRIS_store_ui/issues/53#issuecomment-815506288, the API is self-documenting in some ways and it always links to itself.

In general, if something can be done server-side, it's better to do it server-side. Reasons:

- efficiency: backend has a fast connection to the database, whereas the connection between the browser and backend is a bottleneck. Also backend can use <O(n) algorithms whereas the front-end, to respect the bandwidth problem, must use linear search.
- speed: page loads faster
- battery: JavaScript burns phone batteries, and web browsing on mobile devices makes up around half of users (maybe not for our use case but it's still important)
